src/harness/sas_testcase.py

```diff
@@ -45,3 +53,50 @@ def assertValidResponseFormatForApprovedGrant(self, grant_response):
     # operationParam should not be set if grant is approved
     self.assertFalse('operationParam' in grant_response)
     self.assertTrue(grant_response['channelType'] in ('PAL', 'GAA'))
+
+  def assertRegistered(self, registration_request,
+                       conditional_registration_data=None):
+    for device in registration_request:
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+    if conditional_registration_data:
+      self._sas_admin.PreloadRegistrationData(conditional_registration_data)
+
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+
+    # Check the registration response; collect CBSD IDs
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      self.assertTrue('cbsdId' in resp)
+      cbsd_ids.append(resp['cbsdId'])
+
+    # Return list of cbsd_ids
+    return cbsd_ids
+
+  def assertRegisteredAndGranted(self, registration_request, grant_request,
+                                 conditional_registration_data=None):
+    self.assertEqual(len(registration_request), len(grant_request))
+    cbsd_ids = self.assertRegistered(registration_request,
+                                     conditional_registration_data)
+    list_of_cbsd_ids_and_grant_ids = []
+
+    for cbsd_id, grant_req in zip(cbsd_ids, grant_request):
+      grant_req['cbsdId'] = cbsd_id
+
+    request = {'grantRequest': grant_request}
+    grant_response = self._sas.Grant(request)['grantResponse']
+
+    # Check the grant response; collect CBSD and Grant IDs
+    for cbsd_id, grant_resp in zip(cbsd_ids, grant_response):
```

Comment:
I think this is part of a larger philosophical question: do we want to do light or heavy checks of responses within the helper functions? I can see arguments in both directions:
Pro light: we've already validated the functionality more thoroughly in other tests; let's assume that this works and not double-check all of the time. If it *does* fail, it may increase debugging time. Heavy checking also means that if you have an error in e.g. requesting measurements you may fail tons of unrelated tests; it might be nice to (for the most part) only fail a test if you're missing the functionality that it's testing. Finally, heavy checking has the potential to complicate the implementation of the helper functions, possibly introducing subtle bugs (this argument is much weaker than the rest).
Pro heavy: the SAS should always be right, so the checks should always pass (and if they fail, that's super interesting). By testing more completely, we have higher confidence in SAS performance.

I think in general I'm in favor of light testing for now and possibly adding the heavy testing later. But if you feel strongly in favor of heavy testing now, we can do that instead. Let me know, and let's adjust the comments to reflect the decision so it's clear to the caller.